### PR TITLE
Medical - Fix hanging when vehicle crew receive fatal damage

### LINKED ISSUE #9674: ACE Medical FFV Damage Hang
**Mods (complete and add to the following information):**
- **Arma 3:** `2.14 Stable`
- **CBA:** `3.16.1 Stable`
- **ACE3:** `3.16.1 Stable`

**Description:**
When hitting units in FFV with large/overkill amounts of damage (i.e., main tank gun), Arma will hang for a few seconds.

**Steps to reproduce:**

1. Launch attached demo mission. 
1a. Alternatively, in the editor place a units in FFV seats that are easily hit and a vehicle with a large caliber controllable gun.
2. Shoot through FFV units with large caliber round (i.e., AP tank rounds)
3. Experience a few seconds of hang

**Expected behavior:**
Normally I'd expect minimal overhead from ACE medical

**Where did the issue occur?**
- Dedicated MP & Editor (Singleplayer)

**Log Files:**
https://pastebin.com/xMspFEVE
Also in attached zip.

**Additional context:**
[Demo mission and rpt zip](https://github.com/acemod/ACE3/files/13623134/Demo.mission.and.rpt.zip)

This does not occur with infantry not in a vehicle, but will for those in an FFV seat. I did not check normal passenger seats. Appears to be related to `ace_medical_damage_fnc_woundsHandleBase` or the medical state machine, see screenshot below using current steam versions of intercept and ASP to take a look.

**Screenshots:**
image-1

https://github.com/acemod/ACE3/assets/11843718/b810936e-33a1-4198-a434-6335d30a42aa

### PR DESCRIPTION
**Pull request by LinkIsGrim**
**When merged this pull request will:**
- Title.
- Close #9674 

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

**Comment by veteran29** [comment]
IMO lacks an explaination in code/comment why it's done the way it's done. I guess HD fires many amount of times for single hit and it ended with long array and hashmap allows us to handle every hitpoint once?

**Comment by LinkIsGrim** [comment]
For every 0.1666... structural damage a new entry would be added to the damage array which could result in hundreds or thousands of iterations in woundsHandlerBase in cases such as the one in #9674. This fix is slower for lower damage events but much faster for extreme cases.

TL;DR: Yes, exactly what you said.

**Review by veteran29 — APPROVED** [review]
